python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py

```diff
@@ -345,36 +346,50 @@ def execute_k8s_job(
 
     timeout = timeout or 0
 
-    api_client.wait_for_job(
-        job_name=job_name,
-        namespace=namespace,
-        wait_timeout=timeout,
-        start_time=start_time,
-    )
+    try:
+        api_client.wait_for_job(
+            job_name=job_name,
+            namespace=namespace,
+            wait_timeout=timeout,
+            start_time=start_time,
+        )
+    except (DagsterExecutionInterruptedError, Exception) as e:
```

Comment:
instead of having this try/except block 3 times, should we just have one that cover from job creation until job success? Currently If we got an interrupt during the uncovered parts like log tailing we would not clean-up the job.

Reply:
Yeah, I agree with that, changed.